File: python/leetcode/majority-element.py
class Solution:
    def majorityElement(self, nums):
        seen = {}
        if len(nums) == 1:
            return nums[0]
        majority_count = len(nums) // 2
        for n in nums:
            if n in seen:
                seen[n] += 1
                if seen[n] > majority_count:
                    return n
            else:
                seen[n] = 1

File: java/leetcode/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
	    HashMap<Integer, Integer> seen = new HashMap();
	    if (nums.length == 1) { return nums[0]; }
	    int majorityCount = ~~(nums.length / 2);
	    for (int i = 0; i < nums.length; i++) {
	        if (seen.containsKey(nums[i])) {
	            seen.put(nums[i], seen.get(nums[i]) + 1);
	            if (seen.get(nums[i]) > majorityCount) { return nums[i]; }
	        } else { seen.put(nums[i], 1); }
	    }
        return 0;
    }
}
